### ISSUE TITLE
denominator of nt_xent_loss

### ISSUE THREAD

**Issue by bokyeong1015** [lucidrains/contrastive-learner#2]
Hi, thank you for providing your nice implementation!

I have a minor question on `nt_xent_loss`. I think the loss should be divided by **2b** instead of **2(b-1)** in [here](https://github.com/lucidrains/contrastive-learner/blob/master/contrastive_learner/contrastive_learner.py#L64) to consider the number of samples used in computing the cross entropy loss.
Current: `loss /= 2 * (b - 1)`
Suggestion: `loss /= 2 * b`

What do you think of it? Please correct me if I am wrong.

===
cf. I would like to share my understanding on your code.

`queries` = [A, B] # features from two images A and B, size b x f (here b=2, f=feature dimension)
`keys` = [A’, B’] # features from their augmentation views, size b x f
`logits` in [line 56](https://github.com/lucidrains/contrastive-learner/blob/master/contrastive_learner/contrastive_learner.py#L56): similarity matrix, size 2b x 2b

- AA AB AA’ AB’
- BA BB BA’ BB’
- A’A A’B A’A’ A’B’
- B’A B’B B’A’ B’B’

`logits` in [line 60](https://github.com/lucidrains/contrastive-learner/blob/master/contrastive_learner/contrastive_learner.py#L60): removed diagonals and normalized by temperature, size **2b x (2b-1)**

-   AB **AA’** AB’
-   BA BA’ **BB’**
-   **A’A** A’B A’B’
-   B’A **B’B** B’A’

`labels` in [line 62](https://github.com/lucidrains/contrastive-learner/blob/master/contrastive_learner/contrastive_learner.py#L62): [1, 2, 0, 1] # GT labels for the above logits (indicated in bold), size 2b x 1
`loss` = F.cross_entropy(logits, labels, reduction='sum') # (2b-1)-way classification -> the **2b** loss outputs are summed
`loss` /= 2 * (b - 1) # why?

**Comment by lucidrains** [comment]
@bokyeong1015 ohh, this is an error on my part, thanks for catching this! https://github.com/lucidrains/contrastive-learner/actions/runs/646834849
